docs/madr/decisions/034-default-target-ref-policy.md

```diff
@@ -0,0 +1,104 @@
+# TargetRef policies as a default
+
+* Status: accepted
+
+Technical Story: none
+
+## Context and Problem Statement
+
+At present, Kuma has two policy types:
+
+* Legacy policies with Source and Destination attributes
+* New policies based on TargetRef
+ 
+The legacy policies currently serve as the defaults. However, starting from release 2.6, the intention is to switch the default policy to TargetRef.
+
+## Considered Options
+
+1. Introduce a `policyEngine` field within the `Mesh` object, allowing the selection of the default policy version.
+
+## Decision Outcome
+
+The decision is to introduce a `policyEngine` field within the Mesh object to select the default policy version.
+
+### Positive Consequences
+
+* New deployments will start with the new default TargetRef policies.
+* Kuma remains capable of working with and applying old policies.
+* Existing meshes remain unaffected by the change.
+
+### Negative Consequences
+
+* Possibility of an additional, potentially unused field in the mesh object in the future.
+
+## Pros and Cons of the Options
+
+### Introduce a `policyEngine` field within the `Mesh` object, allowing the selection of the default policy version.
+
+
+The proposed addition to the Mesh object would appear as follows in the protobuf definition:
+
+```protobuf
+// Mesh defines configuration of a single mesh.
+message Mesh {
+
+  // ...
+    
+  enum PolicyEngine {
+    // An Undefined mode means that the policy engine wasn't defined.
+    // When creating, the default is set to TargetRef.
+    // During retrieval of old settings, distinguishing it as Legacy.
+    Undefined = 0;
+    // A Legacy mode creates old default policies.
+    Legacy = 0;
+    // A TargetRef mode creates new default targetRef policies.
+    TargetRef = 1;
+  }
+
+  // policyEngine implies which policy engine should be used as a default.
```

Comment:
Is it only about defaults or is it about which policies are used?

Let's say I've set policyEngine to targetRef. Are old policies still applied?
If so, is "how to migrate to new policies" a part of the current "policy 2.0" effort as well?

Reply:
If this is only abou initial policies then I think we need more explicit name. We already have `skipCreatingInitialPolicies`. Should we name this `initialPoliciesEngine`?

Reply:
But it's not only about initial policies but also about the flow of the code. In some cases, we need to verify which version you want to use and change the flow

Reply:
MeshHTTPRoute/MeshTCPRoute we don't need them so we need to remove check if policy exists, but for old engine we don't want to configure it again

Reply:
I feel like `policyEngine` is a too generic field.

Doc "implies which policy engine should be used as a default" also does not explain much if I'm unfamiliar with the MADR and the discussion.

If I were to guess what will happen, setting this to targetRef would just ignore old policies. But what it does is:
* When mesh is created, create MeshTimeout, MeshRetry, and MeshCircuitBreaker policies instead of old policies
* If TrafficRoute is missing, let the traffic flow
* If TrafficPermission to external is missing (+ no egress with mtls), let the traffic flow

Maybe we need something more explicit
```
apiVersion: kuma.io/v1alpha1
kind: Mesh
metadata:
  name: default
spec:
  initialPoliciesEngine: sourcesDestinations # | targetRef. "sourcesDestinations" by default if missing
  routing:
    allowOnMissingPolicies: false # | true. "false" by default
```

We can document that if you want to only use new policies, create such mesh
```
apiVersion: kuma.io/v1alpha1
kind: Mesh
metadata:
  name: default
spec:
  initialPoliciesEngine: targetRef
  routing:
    allowOnMissingPolicies: true
```

Eventually, in X versions we will remove those fields.

I think introducing `KUMA_DEFAULTS_USE_LEGACY_POLICY_ENGINE` does not add much because you need to set this on every zone and global cp (also multitenant globalcp).
I think setting it in the mesh is more explicit and easier to understand.

Reply:
side q: wasn't there an idea by @michaelbeaumont that there shouldn't be default policies that could break traffic flow? what was our decision there?

Reply:
I've added a change based on the conversation we had. We will not create any default policy and add upgrade note information for users so they can create them by themselves.